gdal/frmts/jpegxr/jpegxrdataset.cpp

```diff
@@ -0,0 +1,1202 @@
+    JPEGXREncoderConfig config; // sets sensible defaults
+
+    if (CSLFetchNameValue(papszOptions, "QUALITY") != NULL)
+    {
+        int n = atoi(CSLFetchNameValue(papszOptions, "QUALITY"));
+        if (n < QUALITY_LOWEST || n > QUALITY_LOSSLESS)
+        {
+            CPLError(CE_Failure, CPLE_IllegalArg,
+                "QUALITY=%s is not a legal value in the range %d-%d.",
+                CSLFetchNameValue(papszOptions, "QUALITY"), QUALITY_LOWEST, QUALITY_LOSSLESS);
+            return nullptr;
+        }
+        config.SetQuality(n);
+    }
+
+    if (CSLFetchNameValue(papszOptions, "OVERLAP") != NULL)
+    {
+        int n = atoi(CSLFetchNameValue(papszOptions, "OVERLAP"));
+        if (OVERLAP_NONE < 0 || n > OVERLAP_COUNT)
+        {
+            CPLError(CE_Failure, CPLE_IllegalArg,
+                "OVERLAP=%s is not a legal value in the range %d-%d.",
+                CSLFetchNameValue(papszOptions, "OVERLAP"), OVERLAP_NONE, OVERLAP_COUNT-1);
+            return nullptr;
+        }
+        config.SetOverlap(n);
+    }
+
+    if (CSLFetchNameValue(papszOptions, "SUBSAMPLING") != NULL)
+    {
+        int n = atoi(CSLFetchNameValue(papszOptions, "SUBSAMPLING"));
+        if (SUBSAMPLING_420 < 0 || n > SUBSAMPLING_COUNT)
+        {
+            CPLError(CE_Failure, CPLE_IllegalArg,
+                "SUBSAMPLING=%s is not a legal value in the range %d-%d.",
+                CSLFetchNameValue(papszOptions, "SUBSAMPLING"), SUBSAMPLING_420, SUBSAMPLING_COUNT-1);
+            return nullptr;
+        }
+        config.SetSubsampling(n);
+    }
+    /* -------------------------------------------------------------------- */
+    /*      Read source data                                                */
+    /* -------------------------------------------------------------------- */
+    int nWordSize = GDALGetDataTypeSize(eDT) / 8;
+    int nUncompressedSize = nXSize * nYSize * nBands * nWordSize;
+    GByte* pabyUncompressedData = (GByte*)VSI_MALLOC_VERBOSE(nUncompressedSize);
+    if (!pabyUncompressedData)
+    {
+        VSIFree(pabyUncompressedData);
+        return nullptr;
+    }
+
+    CPLErr eErr;
+    eErr = poSrcDS->RasterIO(GF_Read, 0, 0, nXSize, nYSize,
+        pabyUncompressedData, nXSize, nYSize,
+        eDT, nBands, NULL,
+        nBands * nWordSize, nBands * nWordSize * nXSize, nWordSize, NULL);
+    if (eErr != CE_None)
+    {
+        VSIFree(pabyUncompressedData);
+        return nullptr;
+    }
+
+    /* -------------------------------------------------------------------- */
+    /*      Setup encoder                                                   */
+    /* -------------------------------------------------------------------- */
+    try
+    {
+        JPEGXREncoder oEncoder;
+        oEncoder.Initialize(pszFilename);
+        oEncoder.SetVerbose(false); // TODO: option?
+        oEncoder.SetSize(nXSize, nYSize);
+        oEncoder.SetResolution(72, 72);
```

Comment:
I've seen 96 used as resolution above. Any particular reason to use 72 rather than anything else ?